core/ustring.cpp

```diff
@@ -3274,20 +3274,22 @@ String String::c_unescape() const {
 	return escaped;
 }
 
-String String::c_escape() const {
+String String::c_escape(bool p_vcs_friendly) const {
 
 	String escaped=*this;
+	if (!p_vcs_friendly) {
+		escaped=escaped.replace("\a","\\a");
+		escaped=escaped.replace("\b","\\b");
+		escaped=escaped.replace("\f","\\f");
+		escaped=escaped.replace("\n","\\n");
+		escaped=escaped.replace("\r","\\r");
+		escaped=escaped.replace("\t","\\t");
+		escaped=escaped.replace("\v","\\v");
+		escaped=escaped.replace("\'","\\'");
+		escaped=escaped.replace("\?","\\?");
+	}
 	escaped=escaped.replace("\\","\\\\");
-	escaped=escaped.replace("\a","\\a");
-	escaped=escaped.replace("\b","\\b");
-	escaped=escaped.replace("\f","\\f");
-	escaped=escaped.replace("\n","\\n");
-	escaped=escaped.replace("\r","\\r");
-	escaped=escaped.replace("\t","\\t");
-	escaped=escaped.replace("\v","\\v");
-	escaped=escaped.replace("\'","\\'");
 	escaped=escaped.replace("\"","\\\"");
-	escaped=escaped.replace("\?","\\?");
 
 	return escaped;
```

Comment:
I think it's not just changing the order here, but actually disabling various escapes when in VCS friendly mode.